From 8ad74cdabac37599d20a985e6a456b1faca92df8 Mon Sep 17 00:00:00 2001
From: Manuel Mommertz <2kmm@gmx.de>
Date: Mon, 9 Jul 2018 14:08:35 +0200
Subject: [PATCH 1/2] configure.ac: Fix --without-caps option

irqbalance links to libcap-ng if it is installed, even when useflag
caps is disabled. This is an error in the configure-script, which
checks for the wrong variable to determine if the user used --with/
--without for libcap-ng.

Downstream bug report: https://bugs.gentoo.org/660782

Signed-off-by: Lars Wendler <polynomial-c@gentoo.org>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index f6c60da..19b148a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -65,7 +65,7 @@ AC_ARG_WITH([libcap-ng],
   AS_HELP_STRING([libcap-ng], [Add libcap-ng-support @<:@default=auto@:>@]))
 
 AS_IF(
-  [test "x$libcap_ng" != "xno"],
+  [test "x$with_libcap_ng" != "xno"],
   [
   PKG_CHECK_MODULES([LIBCAP_NG], [libcap-ng],
     [AC_DEFINE(HAVE_LIBCAP_NG,1,[libcap-ng support])],
-- 
2.18.0

From bc56dbd4eead9ad19ff1d84dc8810123c96503c0 Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Mon, 9 Jul 2018 14:11:44 +0200
Subject: [PATCH 2/2] configure.ac: Fix --enable-irqbalance-ui logic

Before this patch --enable-irqbalance-ui disabled compilation/
installation of the ui because AC_ARG_WITH is used wrong.
The third arguement to AC_ARG_WITH is _action-if-present_ which means
that this action gets triggered once the --enable option has been
submitted to configure on command line no matter if it's enabled or
disabled.
See also https://autotools.io/autoconf/arguments.html chapter 3.1
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 19b148a..75c234d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,9 +43,9 @@ AM_PROG_CC_C_O
 AC_ARG_WITH([irqbalance-ui],
 	[AC_HELP_STRING([--without-irqbalance-ui],
 			[Dont build the irqbalance ui component])],
-			[with_irqbalanceui=false], [with_irqbalanceui=true])
+			[with_irqbalanceui=$withval], [with_irqbalanceui=yes])
 
-AM_CONDITIONAL([IRQBALANCEUI], [test x$with_irqbalanceui = xtrue])
+AM_CONDITIONAL([IRQBALANCEUI], [test x$with_irqbalanceui = xyes])
 
 AC_ARG_WITH([systemd],
   [ AS_HELP_STRING([--with-systemd],[Add systemd-lib support])]
-- 
2.18.0

